Add rule for 'var' keyword (#6)
mizool/playbook: docs/code-conventions/java.md

* In interface method declarations, never use `public` or `abstract` as both are implied.
* Never use `final` on local variables (including method arguments) unless forced by the compiler.

> ### Use of the `var` keyword
> 
> When the variable name is equivalent to the class name (`SplineReticulator splineReticulator = ...`), use of the `var` keyword is encouraged to avoid repetition.
> 
> Additionally, `var` can be used to avoid spelling out the bulky type name of a Lombok-generated builder class.
> 
> However, do not use the `var` keyword in other cases.
> In particular, do not use `var` just because you think the type is "obvious" (as it may not be that obvious to readers including future you).
> Also, avoid using `var` when there are several variables of the same type.  
> 

Reviewer: Perhaps it's just the phrasing or we had a misunderstanding when talking about this but I feel that some of these rules are not what we actually _want_. Let me try to rephrase...

````suggestion
### Use of the `var` keyword

When the variable name is equivalent to the class name (`SplineReticulator splineReticulator = ...`), use of the `var` keyword is encouraged to avoid repetition.
This is also true if there are multiple variables for the same class, that contain the class name:
```java
var destabilizingSplineReticulator = ...;
var branchingSplineReticulator = ...;
```

Additionally, `var` can be used to avoid spelling out the bulky type name of a Lombok-generated builder class.

However, do not use the `var` keyword in other cases.
In particular, do not use `var` just because you think the type is "obvious" (as it may not be that obvious to readers including future you).
Do not use `var` for variables that are return values from method calls (`SplineReticulator splineReticulator = lookupSplineStrategy();`). The method might change and the type of a `var` variable would change with it but the old name might lead to confusion.

````

Author: I discussed this with @codingWombat and @schroenser in person, and we decided to loosen the rule to cover more desirable uses. The gist of it will be "do use `var` (only) where it aids readability, as agreed by your reviewer. here are some examples where `var` really shines."
